source/auth/tests/connection-string.json

```diff
@@ -24,6 +24,18 @@
         "mechanism_properties": null
       }
     },
+    {
+      "description": "should use the database when the authSource is empty",
```

Comment:
Also add a test with no database: `mongodb://user:password@localhost/?authSource=`

Reply:
I have removed this test and added a new one that makes it invalid when `authSource` is empty.